backend/src/test/java/de/neuefische/backend/service/SignUpServiceTest.java

```diff
@@ -0,0 +1,86 @@
+package de.neuefische.backend.service;
+
+import de.neuefische.backend.dto.LoginDataDto;
+import de.neuefische.backend.dto.UserDto;
+import de.neuefische.backend.security.repository.AppUserRepository;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.server.ResponseStatusException;
+
+
+import java.util.Optional;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class SignUpServiceTest {
+
+    private final PasswordEncoder encoder= new BCryptPasswordEncoder();
+    private final AppUserRepository repo = mock(AppUserRepository.class);
+    private final SignUpService signUpService = new SignUpService(repo, encoder);
+
+    @Test
+    @DisplayName("method signUp should return new userDto, when username does not already exist")
+    void signUpTest() {
+        //Given
+        when(repo.existsById("Lucy")).thenReturn(false);
+        LoginDataDto newUser = LoginDataDto.builder().username("Lucy").password("1234abCD").build();
+
+        // When
+        UserDto actual = signUpService.signUp(newUser);
+
+        // Then
+        assertThat(actual, is(new UserDto("Lucy")));
+    }
+
+    @Test
+    @DisplayName("method signUp should should throw HttpStatus.CONFLICT, when username already exists")
+    void signUpWithExistingUserNameTest() {
+        //Given
+        when(repo.existsById("Lucy")).thenReturn(true);
+        LoginDataDto newUser = LoginDataDto.builder().username("Lucy").password("1234abCD").build();
+
+        // When/ Then
+        assertThrows(ResponseStatusException.class, ()->signUpService.signUp(newUser));
```

Comment:
vll macht es noch sinn, den http status zu überprüfen, also `assertThat(response.getStatusCode(), is(HttpStatus.CONFLICT))`
weil du ihn ja auch oben im display name erwähnst

Reply:
Ich glaub das kann ich hier nicht machen, weil ich ja keine Response hab?